sys/net/network_layer/sixlowpan/lowpan.c

```diff
@@ -1110,7 +1110,7 @@ uint8_t lowpan_iphc_encoding(int if_id, const uint8_t *dest, int dest_len,
             else if (dest_len == 2 &&
                      ipv6_buf->destaddr.uint32[2] == HTONL(0x000000ff) &&
                      ipv6_buf->destaddr.uint16[6] == HTONS(0xfe00) &&
-                     ipv6_buf->destaddr.uint16[7] == *((uint16_t *) dest)) {
+                     ipv6_buf->destaddr.uint16[7] == (dest[1] << 8 | dest[0])) {
```

Comment:
Sure: e.g. https://github.com/RIOT-OS/RIOT/pull/2731/files#diff-a17dee4719c27a6c08fc671ec03e9ea5R241 or https://github.com/RIOT-OS/RIOT/pull/2454/files#diff-c0ab51e027bbe0cfdb3482d3e29d6f34R124

If I would just use the field without the byte-order functions, the compiler would warn me about type errors.